src/main/java/uk/nhs/hee/tis/trainee/reference/mapper/GradeMapper.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2020 Crown Copyright (Health Education England)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.trainee.reference.mapper;
+
+import java.util.List;
+import org.mapstruct.Mapper;
+import uk.nhs.hee.tis.trainee.reference.dto.GradeDto;
+import uk.nhs.hee.tis.trainee.reference.model.Grade;
+
+@Mapper(componentModel = "spring")
+public interface GradeMapper {
+
+  GradeDto toDto(Grade grade);
+
+  List<GradeDto> toDtos(List<Grade> grades);
+
+  Grade toEntity(GradeDto gradeDto);
+
+  List<Grade> toEntities(List<GradeDto> gradeDtos);
```

Comment:
Generic question, are Lists in java immutable? It is preferred to use pass the lists as immutable.

Reply:
List is mutable. Here we just add the Mapper interface, it will be read by the Mapstruct dependency which creates the Mapper class for us after building this project. The list needs to be modified by Mapstruct.

Reply:
Why do we need to modify the input list? Aren't we supposed to read the list and build the output(dto -> entity , entity -> dto).